## Name - Avijit Biswas    Roll No - 2023dss1008

This is the Question link [Click Hear](https://docs.google.com/document/d/1iTIxAUqp99urvgj4BzYfO-X9fiXve6QtBhcuINDTTvI/edit?pli=1)

Question 1.
A manufacturer of leather belts makes three types of belts A, B and C which are processed on three machines M1, M2 and M3. Belt A requires 2 hours on machine M1 and 3 hours on machine M3. Belt B requires 3 hours on machine M1, 2 hours on machine M2 and 2 hours on machine M3. Belt C requires 5 hours on machine M2 and 4 hours on machine M3. There are 8 hours of time per day available on machine M1, 10 hours of time per day available on machine M2 and 15 hours of time per day available on machine M3. The profit gained is Rs. 3/unit, Rs. 5/unit and Rs. 4/unit for A, B and C respectively. Find the daily production of each type for optimal profit.

In [84]:
from scipy.optimize import linprog

In [85]:
# Objective coefficients (profit per unit)
c = [-3, -5, -4]  # Minimize -profit to maximize profit

In [86]:
# Coefficients of the inequality constraints (machine time constraints)
A = [[2, 0, 3],  # Machine M1 constraints for belts A, B, C
     [0, 2, 5],  # Machine M2 constraints for belts A, B, C
     [3, 2, 4]]  # Machine M3 constraints for belts A, B, C

# Right-hand side of the inequality constraints (available hours per day)
b = [8, 10, 15]

# Bounds for the variables (production of each belt type)
bounds = [(0, None), (0, None), (0, None)]

In [87]:
# Solve the linear programming problem
result = linprog(c, A_ub=A, b_ub=b, bounds=bounds, method='highs')

In [88]:
# Print the results
print("Optimal daily production:")
print("Belt A:", round(result.x[0], 2))
print("Belt B:", round(result.x[1], 2))
print("Belt C:", round(result.x[2], 2))

Optimal daily production:
Belt A: 1.67
Belt B: 5.0
Belt C: 0.0


Question 2.
A diet for a sick person must contain atleast 4,000 units of vitamins, 50 units of minerals and 1,400 of calories. Two foods, A and B are available at a cost of Rs. 4 and Rs. 3 per unit respectively. If one unit of A contains 200 units of vitamins, 1 unit of mineral and 40 calories, and one unit of food contains 100 units of vitamins, 2 units of minerals and 40 calories, what combination of foods should be used to have least cost?


In [89]:
# Objective coefficients (cost per unit of food)
c = [4, 3]  # Cost of food A and food B respectively

# Coefficients of the inequality constraints (nutritional requirements)
A = [[-200, -100],  # Vitamins constraint for food A and food B
     [-1, -2],      # Minerals constraint for food A and food B
     [-40, -40]]    # Calories constraint for food A and food B

# Right-hand side of the inequality constraints (minimum nutritional requirements)
b = [-4000, -50, -1400]

In [90]:
# Bounds for the variables (amount of each food)
bounds = [(0, None), (0, None)]

In [91]:
# Solve the linear programming problem
result = linprog(c, A_ub=A, b_ub=b, bounds=bounds, method='highs')

In [92]:
# Print the results
print("Optimal combination of foods:")
print("Units of Food A:", round(result.x[0], 2))
print("Units of Food B:", round(result.x[1], 2))

Optimal combination of foods:
Units of Food A: 5.0
Units of Food B: 30.0


Question 3.
An advertising agency wishes to reach 2 types of audiences: Customers with annual income greater than Rs 60,000, say A, and customers with annual income less than Rs. 60, 000, say B. The total advertising budget is Rs 2,00,000. One programme of TV advertising costs Rs 50,000, one programme of radio advertising costs Rs 20,000. For contract reasons, atleast 3 programmes ought to be of TV, and the number of radio programmes must be limited to 5. Surveys indicate that a single TV programme reaches 4,50,000 customers in category A and 50,000 customers in category B. One radio programme reaches 20,000 customers in category A and 80,000 customers of category B. Determine the media mix to maximize the total reach.


In [93]:
import pulp

In [94]:
# Create a LP problem
prob = pulp.LpProblem("AdvertisingProblem", pulp.LpMaximize)

In [95]:
# Define decision variables
tv_programs = pulp.LpVariable('TV_programs', lowBound=3, upBound=None, cat='Integer')
radio_programs = pulp.LpVariable('Radio_programs', lowBound=0, upBound=5, cat='Integer')

In [96]:
# Define objective function (total)
prob += 450000 * tv_programs + 20000 * radio_programs, "Total Reach"

In [97]:
# Define constraints
prob += 50000 * tv_programs + 20000 * radio_programs <= 200000  # Budget constraint
prob += tv_programs >= 3  # At least 3 TV programs
prob += tv_programs + radio_programs <= 5  # Maximum 5 programs total

In [98]:
prob.solve()

1

In [99]:
print("Optimal TV Programs :", int(pulp.value(tv_programs)))
print("Optimal Radio Programs:", int(pulp.value(radio_programs)))

Optimal TV Programs : 4
Optimal Radio Programs: 0


##### Use the link provided to read on how to perform sensitivity analysis. Choose a case study and perform the same. Write a short report based on your findings

###### A company produces two types of TVs, one is black and white, while the other is colour. The company has the resources to make at most 200 sets a week. Creating a black and white set costs Rs. 2700 and Rs. 3600 to create a coloured set. The business should spend no more than Rs. 648000 a week producing TV sets. If it benefits from Rs. 525 per set of black and white and Rs. 675 per set of colours, How many sets of black/white and coloured sets should it produce in order to get maximum profit?

In [100]:
# Objective coefficients (negative as linprog minimizes the objective)
c = [-525, -675]  # Negative because linprog minimizes the objective

# Coefficients of the inequality constraints
A = [[1, 1],     # Constraint 1: x + y <= 200
     [2700, 3600]]  # Constraint 2: 2700x + 3600y <= 648000

# Right-hand side of the inequality constraints
b = [200, 648000]

In [101]:
# Bounds for the variables
bounds = [(0, None), (0, None)]  # Non-negativity constraint

In [102]:
# Solve the linear programming problem
result = linprog(c, A_ub=A, b_ub=b, bounds=bounds)

In [103]:
# Print the results
print("Optimal production plan:")
print("Black and White sets:", round(result.x[0], 2))
print("Coloured sets:", round(result.x[1], 2))

Optimal production plan:
Black and White sets: 80.0
Coloured sets: 120.0


Objective:
The objective of this analysis is to determine the optimal production plan for a company producing black and white TVs and coloured TVs, aiming to maximize profit within given resource and budget constraints.


Methodology:
We utilized linear programming techniques to model and solve the optimization problem. The objective function was designed to maximize profit, considering the profit earned per set of each TV type. Constraints were imposed on the total number of sets produced per week and the total cost of production per week.

Findings:
After solving the linear programming problem using the linprog function from the scipy.optimize library, the optimal production plan was determined as follows:

Black and White sets: 80
Coloured sets: 120<
Sensitivity Analysis:
We conducted sensitivity analysis to understand the impact of changes in key parameters on the optimal production plan:

Profit Per Set of Black and White TV: Increasing the profit per set of black and white TV from Rs. 525 to Rs. 600 resulted in an adjustment of the optimal production plan, with fewer black and white sets produced and more coloured sets produced.
Budget Constraint: Raising the budget constraint from Rs. 648,000 to Rs. 700,000 led to a change in the optimal production plan, with an increase in the production of both black and white sets and coloured sets.
Shadow Values:
Shadow values represent the rate of change in the objective function value with respect to a unit change in the right-hand side of the constraints. In our analysis:

For the constraint on the total number of sets produced per week, the shadow value indicates the change in profit for a one-unit increase in the production limit.
For the budget constraint, the shadow value denotes the change in profit for a one-unit increase in the budget limit.





Summary:
The optimized production plan of 80 black and white sets and 120 coloured sets ensures efficient resource utilization and maximizes profitability within the given constraints. Sensitivity analysis highlights the flexibility of the production plan in response to changes in profit margins and budget constraints, providing valuable insights for strategic decision-making.

## Sensitivity Analysis has been parformed in Excel of Question No 1. To find [Click hear](https://docs.google.com/spreadsheets/d/1oWSYuM8snFMV9WPUc4MyYJo_p5rlZ7Hc/edit?usp=sharing&ouid=106200472672383295712&rtpof=true&sd=true)